Rewrite of Using Radicle section to capture feedback from onboarding (#49)
radicle-dev/radicle-docs: docs/using-radicle/fetching-and-merging.md

and merge contributions from other people. To fetch and merge changes from a
contributor, you first need to add them as a remote to your project. If you
haven't done so, then first follow the steps in [Tracking & viewing
contributions][tv] section.

If you can see your a tracked peer's changes after adding them as a remote, this
means they have been successfully replicated to your machine. If you have added
a peer as a remote and can't see their changes, head over to our
[Troubleshooting][tr] section.

## Fetching changes 

If you're interested in merging these changes into one of your branches, you'll
need to fetch these commits into your working copy. To make sure you can fetch or push changes, you need to add the
[`git-rad-remote` helper][gs] to your `$PATH`.

Navigate to your woking copy in a terminal. Here you can inspect the remotes of
your project by running the `git remote` command. For example, a project that
tracks multiple people might look something like this.

```
$ git remote show
fintohaps@hybaapjdrhj8mg4mathpwjaeuktfzxd8uqfpjknp65fyncehpxxzze
merle@hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g
rad

> xla@hydxmk8cx8mcbijp6kgb8htm6jdhpstiow3kbezxp438adybaaqzgy

Reviewer: ```suggestion
```

Loosk messy otherwise and the message still comes across. Thanks for including me. :>

```

As you can see, the format of a remote handle is:

```
<display name>@<Device ID>
```

> where the display name is the name you see on a user's profile page in Upstream.

Reviewer: We don't have profile pages yet, maybe we mention the peer/remote selector here as this is what the user will have encountered already.

Author: ```suggestion
where the display name is the name you see when you add a Device ID to your peer selector in Upstream.
```

To fetch the changes of a specific remote, run the `git fetch` command with
their remote handle:

```
$ git fetch merle@hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g
remote: Enumerating objects: 7, done.
remote: Counting objects: 100% (7/7), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 6 (delta 0), reused 6 (delta 0), pack-reused 0
Unpacking objects: 100% (6/6), 1.50 KiB | 1.50 MiB/s, done.
From rad://hwd1yre8wtxufyqnt8xzxt8ps3n7ww8qzpfmud5pibarumshawnpcrjzjme
 * [new ref]         hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g/heads/main -> merle/main
```

A successful fetch will have a similar output to the example above and show that
the reference is now located as a remote branch (e.g. `merle@hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g/main`). We can
confirm this is under a remote branch by entering the following command:

```
$ git branch -a
main
remotes/merle@hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g/main
remotes/rad/main
```

Once changes have been fetched, you can view the changes by getting a diff of
your changes:

```

> $ git diff merle/main 

Reviewer: ```suggestion
$ git diff merle@hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g/main 
```

diff --git a/README.md b/README.md
index b72e7d4..8c98464 100644
--- a/README.md
+++ b/README.md
@@ -1 +1 @@
-Radicle is the next big thing!
+Radicle
```

If you are happy with these changes, you can merge them into your own branch:

```

> $ git merge merle/main

Reviewer: ```suggestion
$ git merge merle@hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g/main
```

[main e6ee76d] Merge remote-tracking branch 'merle/main' into main
```

or for those of you who prefer rebasing:

```

> $ git rebase merle/main

Reviewer: ```suggestion
$ git rebase merle@hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g/main
```

[main 3e0e32d] Radicle Update
 1 file changed, 2 insertions(+)
Successfully rebased and updated refs/heads/main.
```

This puts your working copy in a state this up-to-date with your contributor's
changes. 

You can publish these changes to the Radicle network by running `git push
rad`(See [Pushing changes][pc]):

```
$ git push rad main
Enumerating objects: 2, done.
Counting objects: 100% (2/2), done.
Delta compression using up to 8 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (2/2), 1.72 KiB | 1.72 MiB/s, done.
Total 2 (delta 0), reused 0 (delta 0), pack-reused 0
To rad://hwd1yre8wtxufyqnt8xzxt8ps3n7ww8qzpfmud5pibarumshawnpcrjzjme.git
   e6ee76d..57917bf  main -> main
```

Once you have published the changes from your working copy, you can view the
changes you made in **Radicle Upstream**. If the changes were successfully

> published to the network, you should see the new commits on your project page.

Reviewer: They are only published if you are online, and you can still see them when offline.

Author: Fixed this 👍

actions. To read more about how Radicle is built on Git see our [How it
Works][gi] section.

To fetch the changes of a specific remote, run the `git fetch` command with
their remote handle:

```
$ git fetch merle@hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g
remote: Enumerating objects: 7, done.
remote: Counting objects: 100% (7/7), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 6 (delta 0), reused 6 (delta 0), pack-reused 0
Unpacking objects: 100% (6/6), 1.50 KiB | 1.50 MiB/s, done.
From rad://hwd1yre8wtxufyqnt8xzxt8ps3n7ww8qzpfmud5pibarumshawnpcrjzjme
 * [new ref]         hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g/heads/main -> merle/main
```

A successful fetch will have a similar output to the example above and show that
the reference is now located as a remote branch (e.g. `merle/main`). We can
confirm this is under a remote branch by entering the following command:

```
$ git branch -a
main
remotes/merle/main
remotes/rad/main
```

<details>
<summary>**Where did the
`@@hyb1exienq5txww75rgdftuf6j78qehtigma6hn19zt8gz5pu5k73g` go?**</summary>
The suffix of the remotes is disappearing and is a known

> [issue](https://github.com/radicle-dev/radicle-upstream/issues/1389).

Reviewer 2: @MeBrei once your fix lands could you update this section please? :bow:

Reviewer 3: :+1: Will add it to the issue!

Reviewer 3: removed/updated in 12b1094

Reviewer 2: It must have been fate that I had you in the copy and you were the one to fix it :smile:
